controllers/istio/istio_controller.go

```diff
@@ -78,7 +79,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, istio *v1.Istio) (ctrl.Resul
 	log.Info("Reconciliation done. Updating status.")
 	statusErr := r.updateStatus(ctx, istio, reconcileErr)
 
-	return result, errors.Join(reconcileErr, statusErr)
+	return result, errors.Join(reconcileErr, statusErr, errors.Unwrap(reconcileErr))
```

Comment:
I'm not sure why all reconcilers now also return the unwrapped errors? it would be simpler if the `Error()` method simply made sure that any `originalError` is also reported to the user